src/AggregateFunctions/AggregateFunctionsArgMinArgMax.cpp

```diff
@@ -224,12 +251,82 @@ class AggregateFunctionArgMinMax final
     }
 };
 
+/// If we will add more it will compile forever
+using TypesToCreateSpecializedData = std::tuple<UInt8, UInt16, UInt32, UInt64, Int8,
```

Comment:
I think you can make a check `isNativeUInt() || isNativeInt() || isNativeFloat() || isString()` and then jump into "native" types using `TypeIndexToType`. This will remove additional template function instantiations.